### ISSUE TITLE
FB SDK version Needs upgrade

### ISSUE THREAD

**Issue by userlab-luisdeleon** [thebergamo/react-native-fbsdk-next#476]
## 🐛 Bug Report
in latest version
`react-native-fbsdk-next: 12.1.1 `

![image](https://github.com/thebergamo/react-native-fbsdk-next/assets/59143067/dcf402a6-14ed-4680-acd6-2013bebd76f8)

I have installed the latest version when I integrate it and check the FB console I get this message
any ideas?

**Comment by AronBe** [comment]
Seeing the same

**Comment by mikehardy** [comment]
I think someone needs to propose a PR that integrates the latest SDK and confirm they have tested it and it works.
It's open source, you are a fully empowered developer, not just a powerless user waiting for someone to help you.
I'm happy to merge and release any reasonable PR if you can show you've tested it and it works

**Comment by jamesrogers93** [comment]
I'm seeing the same issue. Annoyingly this is blocking us from running Facebook ads.

A temporary fix might be to downgrade to use v11.3.0 as it uses FBSDK v15. (I believe this is a bug with FBSDK v16).

Will let you know if this works for me.

**Comment by mikehardy** [comment]
@jamesrogers93 or....

> I think someone needs to propose a PR that integrates the latest SDK and confirm they have tested it and it works.

**Comment by jamesrogers93** [comment]
I suggested it as a temporary fix, not a permanent fix.

**Comment by mikehardy** [comment]
No one wants to do the work to try to move things forward 😆 
Not picking on you specifically, this is the tragedy of open source encapsulated in one single issue
Everyone interested, no one moving it forward. Eventually someone will *really* need it then it'll move

**Comment by CedYF** [comment]
still have this issue

**Comment by mikehardy** [comment]
Still comments that don't advance anything but they generate notifications for everyone, smiley

**Comment by 4yki** [comment]
Seems like there is an issue with native facebook-android-sdk https://github.com/facebook/facebook-android-sdk/issues/1191 
No clear answers from contributors yet... Anyway there is nothing to do with RN package. I think this issue must be closed as it's not an actual library issue.
I don't use much functionality (just events tracking) so for me simple patch with downgraded SDK versions works. Not sure if it affects other features.

**Comment by vyancharuk** [comment]
@jamesrogers93 @alzalabany Is this issue still present in the latest SDK version, v13.0.0 (FB v17)?

**Comment by Anas-7** [comment]
@vyancharuk Yes it is still present. Downgrading to 11.3.0 fixed the Android issue but iOS is still stuck at 0.0.2

**Comment by i-ked-ii** [comment]
Now needs upgrade more than v17.0.0

![Screenshot 2567-09-10 at 10 23 47](https://github.com/user-attachments/assets/fd8573d0-ab69-44f6-8747-4bb6b8933ab7)

**Comment by hakkikonu** [comment]
_Claude response:_ 

Why the "Update now" banner still shows
Multiple independent reports confirm the dashboard lies:

* It's an upstream dashboard bug tracked in react-native-fbsdk-next issue [#476](https://github.com/thebergamo/react-native-fbsdk-next/issues/476) — "FB SDK version Needs upgrade (upstream issue it seems, does not seem to affect functionality)". The maintainer (mikehardy) and several users confirmed the banner does NOT correlate with what SDK the app actually ships; downgrading to an older SDK and then back has even been reported to clear it. It's pinned in the repo with label upstream because there is nothing the RN wrapper can do — it's on Meta's side.

* The dashboard reads the SDK version from the next inbound event, not from the build you've released. From Stack Overflow question [ios-facebook-sdk-event-manager-requiring-update-fb-sdk](https://stackoverflow.com/questions/75999948/ios-facebook-sdk-event-manager-requiring-update-fb-sdk) and [updating-fb-sdk-to-latest-version-not-working](https://stackoverflow.com/questions/55040745/updating-fb-sdk-to-latest-version-not-working): **_Meta only updates the displayed SDK version after it receives a fresh app event from a released build running the new SDK. Test Events from Xcode don't always count, and TestFlight builds are sometimes ignored too._**

* Dashboard processing lag. Meta's own support forum thread ([developers.facebook.com/community/threads/874466431444565](https://developers.facebook.com/community/threads/874466431444565/)) has developers reporting the banner staying for 3–14 days even after a fully released production build started sending events. Accepted SO answer: "after 3 days passed, fb allow me create adv campaign. But it still shows me warning I need update SDK." → cosmetic only.

How to confirm you're actually OK
Since the banner is decorative at this point, verify the real integration with signals the dashboard will eventually catch up to:

* Check the SDK version at runtime — add a one-liner console.log(Settings.getSDKVersion?.()) after initializeSDK() in _layout.tsx. It should print 18.0.3.
* Ship a release/TestFlight build, then in Events Manager → Test Events tab send an event from that installed build using a test device code. If events appear there with the right parameters, attribution is working — the outdated warning on the Overview tab is just a stale cache.
* Watch the All Activity graph in your screenshot — the declining line is a separate attribution/iOS 14.5+ issue (SKAdNetwork + ATT denial reduces signals), not an SDK-version issue. After FBSDK 18 ships in your production build for a few days, the "Update now" pill generally drops off on its own.

Practical recommendation
* Don't chase this banner. Per the pinned RN wrapper issue and multiple SO / developer-forum threads, the consensus is: no action needed if the binary truly links FBSDK 18+ (yours does per Podfile.lock).
* Do fix the deprecation warning from before, since that's the one in your code: drop Settings.setAdvertiserTrackingEnabled on iOS 17+ (it's a no-op now — SDK reads ATTrackingManager.trackingAuthorizationStatus directly). See Meta's current doc at [Advertising Tracking Enabled](https://developers.facebook.com/docs/app-events/guides/advertising-tracking-enabled/).
* Give it a week after your next production release. If the banner is still there after your app is live with confirmed events flowing, it's safe to ignore — many shipped apps run ads fine with the banner still showing.
Sources: react-native-fbsdk-next [#476](https://github.com/thebergamo/react-native-fbsdk-next/issues/476) (pinned), facebook-android-sdk [#1191](https://github.com/facebook/facebook-android-sdk/issues/1191), Meta [iOS Upgrade Guide](https://developers.facebook.com/docs/ios/upgrade-guide/), Stack Overflow threads above.
